src/RoboSaverVirtualModule.sol

```diff
@@ -181,24 +203,55 @@ contract RoboSaverVirtualModule {
     /// @param _amount The amount of $EURe to deposit or withdraw
     function adjustPool(PoolAction _action, uint256 _amount) external onlyKeeper {
         if (_action == PoolAction.WITHDRAW) {
-            _poolWithdrawal(CARD, _amount);
+            /// @dev Close the pool in case the $EURe available for withdrawal is less than the deficit
+            uint256 withdrawableEure =
+                BPT_STEUR_EURE.balanceOf(CARD) * BPT_STEUR_EURE.getRate() * (MAX_BPS - slippage) / MAX_BPS;
+            if (withdrawableEure < _amount) {
+                _poolClose(withdrawableEure);
+            } else {
+                _poolWithdrawal(_amount);
+            }
         } else if (_action == PoolAction.DEPOSIT) {
-            _poolDeposit(CARD, _amount);
+            _poolDeposit(_amount);
         }
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                                    INTERNAL METHODS
     //////////////////////////////////////////////////////////////////////////*/
 
+    /// @notice Close the pool position by withdrawing all to $EURe
+    /// @param _minAmountOut The minimum amount of $EURe to withdraw from the pool
+    /// @return request_ The exit pool request as per Balancer's interface
+    function _poolClose(uint256 _minAmountOut) internal returns (IVault.ExitPoolRequest memory request_) {
+        /// @dev All asset related arrays should always follow this (alphabetical) order
+        IAsset[] memory assets = new IAsset[](3);
+        assets[0] = IAsset(address(STEUR));
+        assets[1] = IAsset(address(BPT_STEUR_EURE));
+        assets[2] = IAsset(address(EURE));
+
+        /// @dev Allow for one wei of slippage
+        uint256[] memory minAmountsOut = new uint256[](3);
+        minAmountsOut[2] = _minAmountOut - 1;
+
+        /// @dev The `exitTokenIndex` for $EURe is 2
+        bytes memory userData =
```

Comment:
thinking maybe its worth to make any "magic number" around the contract to make them its own constant 

i.e: 

```
EURE_TOKEN_POOL_INDEX = 2
```

Reply:
maybe there is a way to optimise the double occurrence of `BPT_STEUR_EURE.balanceOf(CARD)` while going the pool closing flow:

`one happens in L208, then again in L239`

minor gas optimisation if possible

Reply:
i introduced some more constants in a3ebf57. have a look though, i might have done something weird with creating an array 'constant'

Reply:
going to leave the optimisation for now, unless it occurs in even more places

Reply:
moved it to a private storage section in bbd95ca9d0b05209ababef79866adf14df88f7a7